src/datasources/db/v2/postgres-database.module.ts

```diff
@@ -1,8 +1,20 @@
 import { Module } from '@nestjs/common';
 import { PostgresDatabaseService } from '@/datasources/db/v2/postgres-database.service';
+import { DatabaseMigrator } from '@/datasources/db/v2/database-migrator.service';
+import { ConfigModule } from '@nestjs/config';
+import { DatabaseShutdownHook } from '@/datasources/db/v2/database-shutdown.hook';
+import { DatabaseInitializeHook } from '@/datasources/db/v2/database-initialize.hook';
+import { DatabaseMigrationHook } from '@/datasources/db/v2/database-migration.hook';
 
 @Module({
-  providers: [PostgresDatabaseService],
-  exports: [PostgresDatabaseService],
+  imports: [ConfigModule],
+  providers: [
+    PostgresDatabaseService,
+    DatabaseMigrator,
+    DatabaseInitializeHook,
+    DatabaseShutdownHook,
+    DatabaseMigrationHook,
+  ],
+  exports: [PostgresDatabaseService, DatabaseMigrator],
```

Comment:
Can be done in a followup, but if we're adding a suffix to the module, perhaps we should also to all providers as well?

Reply:
I generalized a bit here. The goal was to create a migrator capable of migrating different databases, but the `PostgresDatabaseService` only handles `PostgreSQL`. For example, we can have:

`DatabaseMigrator`
`PostgresDatabaseService`
`MongoDatabaseService`

Reply:
It was more in reference to the -v2 suffix. It could be later removed. If you think this won't be touched often then we can keep it as is.

Reply:
Oh, sorry, I read it as 'prefix'. Personally, I prefer not to add a suffix, as we typically don't register different versions of a module. However, I've added the suffix here until we deprecate v1. What do you think?

Reply:
I don't _think_ we'll import these providers often but I think it'd be beneficial to add the suffix for now and, as you said, remove it when we deprecate v1.

Reply:
I suggest we don't add suffix as the module is the face of this component and needs a suffix but the providers are all internally registered so for that reason I believe we don't need suffix. What do you think?

Reply:
Sounds good to me.